src/core/package_cache.cpp

```diff
@@ -170,45 +179,42 @@ namespace mamba
                     }
 
                     // Validate checksum
-                    if (valid)
+                    if (!s.sha256.empty())
                     {
-                        if (!s.sha256.empty())
+                        if (s.sha256 != repodata_record["sha256"].get<std::string>())
                         {
-                            if (s.sha256 != repodata_record["sha256"].get<std::string>())
-                            {
-                                valid = false;
-                                LOG_WARNING << "Extracted package cache '" << extracted_dir.string()
-                                            << "' has invalid SHA-256 checksum";
-                            }
-                            else if (s.size == 0)
-                            {
-                                // in case we have no s.size
-                                // set valid true here
-                                valid = true;
-                            }
+                            valid = false;
+                            LOG_WARNING << "Extracted package cache '" << extracted_dir.string()
+                                        << "' has invalid SHA-256 checksum";
                         }
-                        else if (!s.md5.empty())
+                        else if (s.size == 0)
                         {
-                            if (s.md5 != repodata_record["md5"].get<std::string>())
-                            {
-                                LOG_WARNING << "Extracted package cache '" << extracted_dir.string()
-                                            << "' has invalid MD5 checksum";
-                                valid = false;
-                            }
-                            else if (s.size == 0)
-                            {
-                                // for explicit env, we have no size, nor sha256 so we need to
-                                // set valid true here
-                                valid = true;
-                            }
+                            // in case we have no s.size
+                            // set valid true here
+                            valid = true;
```

Comment:
looks like this condition is not used anymore

Reply:
this condition is still used if the repodata_record.json doesn't contain a value (== "").

Reply:
I don't get it, default value is already set to `true`